prover/src/matrix/row_matrix.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use super::segments::Segment;
+use crate::{matrix::ARR_SIZE, Matrix};
+use math::{fft, FieldElement, StarkField};
+use utils::collections::Vec;
+use utils::{flatten_vector_elements, uninit_vector};
+
+// ROWMAJOR MATRIX
+// ================================================================================================
+
+/// A row-major matrix of field elements. The matrix is represented as a single vector of field
+/// elements, where the first `row_len` elements represent the first row of the matrix, the next
+/// `row_len` elements represent the second row, and so on.
+///
+/// # Note
+/// - The number of rows in the matrix is always a multiple of the ARR_SIZE.
+/// - The number of columns in the matrix is always a multiple of the ARR_SIZE.
```

Comment:
nit, I'd remove "the"
```suggestion
/// - The number of rows in the matrix is always a multiple of ARR_SIZE.
/// - The number of columns in the matrix is always a multiple of ARR_SIZE.
```